server/src/main/java/org/candlepin/controller/mode/ModeChangeReason.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.controller.mode;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+
+import java.util.Date;
+
+
+
+/**
+ * Container class for a suspend mode reason and message pairing
+ */
+public class ModeChangeReason {
+    private final String reasonClass;
+    private final String reason;
+    private final Date time;
+    private final String message;
+
+    /**
+     * Creates a new mode change reason with the given reason and message
+     *
+     * @param reason
+     *  A short string identifying the reason for entering suspend mode
+     *
+     * @param message
+     *  An optional, detailed message providing additional information for the reason Candlepin
+     *  operations are suspended
+     *
+     * @throws IllegalArgumentException
+     *  if the provided reason string is null or empty
+     */
+    public ModeChangeReason(String reasonClass, String reason, Date time, String message) {
```

Comment:
nitpick: reasonClass not documented in javadoc.

Reply:
fixed